akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/AccountExample.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package jdocs.akka.persistence.typed;
+
+import akka.actor.typed.Behavior;
+import akka.actor.typed.javadsl.ActorContext;
+import akka.actor.typed.javadsl.Behaviors;
+import akka.persistence.typed.javadsl.CommandHandler;
+import akka.persistence.typed.javadsl.CommandHandlerBuilder;
+import akka.persistence.typed.javadsl.EventHandler;
+import akka.persistence.typed.javadsl.PersistentBehavior;
+
+public class AccountExample extends PersistentBehavior<AccountExample.AccountCommand, AccountExample.AccountEvent, AccountExample.Account> {
+
+  interface AccountCommand {}
+  public static class CreateAccount implements AccountCommand {}
+  public static class Deposit implements AccountCommand {
+    public final double amount;
+
+    public Deposit(double amount) {
+      this.amount = amount;
+    }
+  }
+  public static class Withdraw implements AccountCommand {
+    public final double amount;
+
+    public Withdraw(double amount) {
+      this.amount = amount;
+    }
+  }
+  public static class CloseAccount implements AccountCommand {}
+
+  interface AccountEvent {}
+  public static class AccountCreated implements AccountEvent {}
+  public static class Deposited implements AccountEvent {
+    public final double amount;
+
+    Deposited(double amount) {
+      this.amount = amount;
+    }
+  }
+  public static class Withdrawn implements AccountEvent {
+    public final double amount;
+
+    Withdrawn(double amount) {
+      this.amount = amount;
+    }
+  }
+  public static class AccountClosed implements AccountEvent {}
+
+  interface Account {}
+  public static class EmptyAccount implements Account {}
+  public static class OpenedAccount implements Account {
+    public final double balance;
+
+    OpenedAccount(double balance) {
+      this.balance = balance;
+    }
+  }
+  public static class ClosedAccount implements Account {}
+
+  public static Behavior<AccountCommand> behavior(String accountNumber) {
+    return Behaviors.setup(context -> new AccountExample(context, accountNumber));
+  }
+
+  public AccountExample(ActorContext<AccountCommand> context, String accountNumber) {
+    super(accountNumber);
+  }
+
+  @Override
+  public Account emptyState() {
+    return new EmptyAccount();
+  }
+
+  private CommandHandlerBuilder2<EmptyAccount> initialHandler() {
```

Comment:
I tried to get rid of the type parameters of the CommandHandlerBuilder. Previously it was
```
private CommandHandlerBuilder<AccountCommand, AccountEvent, EmptyAccount, Account> initialHandler() {
```

By adding `CommandHandlerBuilder2` as non-static inner class of PersistentBehavior it kind of works. Still not sure if there is a problem or not. IntelliJ complains with:
<img width="1126" alt="screen shot 2018-08-22 at 13 40 04" src="https://user-images.githubusercontent.com/336161/44461397-fbdc3380-a610-11e8-9312-a1f464cd74ca.png">

but sbt doesn't warn, although we have `-Xlint:unchecked` in the build

Reply:
Unfortunately I don't think we can get around the "unchecked assignment" warning and then the second commit here must be removed.

Reply:
meaning that we have to revert the second commit here